### ISSUE TITLE
segfault in remotectl certificate if openssl is not installed

### ISSUE THREAD

**Issue by JZXPacky** [cockpit-project/cockpit#8690]
Installed Cockpit on 2 Raspberry Pi 3's with no issues, 1 running Stretch and the other running Jessie. Installed on 3 Pi Zero W's running Jessie and none of them work, all are getting the below.
```
● cockpit.service - Cockpit Web Service
   Loaded: loaded (/lib/systemd/system/cockpit.service; static)
   Active: failed (Result: signal) since Fri 2018-02-23 12:04:41 AEST; 11s ago
     Docs: man:cockpit-ws(8)
  Process: 15521 ExecStartPre=/usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type= (code=killed, signal=SEGV)

Feb 23 12:04:41 PackyPiZeroW01 systemd[1]: cockpit.service: control process exited, code=killed status=11
Feb 23 12:04:41 PackyPiZeroW01 systemd[1]: Failed to start Cockpit Web Service.
Feb 23 12:04:41 PackyPiZeroW01 systemd[1]: Unit cockpit.service entered failed state.
```

Running the command `/usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type=` manually from the shell gives a segmentation fault.

**Comment by martinpitt** [comment]
Which Cockpit version is that? Are you using the official jessie-backports-sloppy? I. e. 160-1~bpo8+1?

Can you please run that command through gdb and get a stack trace? Jessie does not yet have proper debug symbols, but it should at least give some initial idea where things go wrong, and we can go from there.

**Comment by JZXPacky** [comment]
This is how it was installed, using the instructions from [here.](http://cockpit-project.org/running.html#debian) As mentioned, I have it running on both Jessie and Stretch on RPi 3's, but it isn't working on the RPiZeroW's (not that I think that has any relevance, just making note of it).

```
echo 'deb http://deb.debian.org/debian jessie-backports-sloppy main' >  /etc/apt/sources.list.d/backports.list
apt-get update
apt-get install cockpit
```

I'm not sure how I go about running that through gdb sorry.

**Comment by martinpitt** [comment]
Like this:
```
$ sudo gdb --args /usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type=
(gdb) run
[... let it crash ... ]
(gdb) bt full
```
and copy&paste the entire output.

**Comment by JZXPacky** [comment]
Thanks for that.

Ok, here's the output.

```
root@PackyPiZeroW01:~ # gdb --args /usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type=
GNU gdb (Raspbian 7.7.1+dfsg-5+rpi1) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/sbin/remotectl...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type=
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00061782 in ?? ()
(gdb) bt full
#0  0x00061782 in ?? ()
No symbol table info available.
#1  0x000125ce in ?? ()
No symbol table info available.
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb)
```

**Comment by martinpitt** [comment]
Thanks. I'm afraid that backtrace is completely useless (not your fault), so we still don't know what's going on..

**Comment by JZXPacky** [comment]
Well that sucks :(

**Comment by JZXPacky** [comment]
So I'm assuming that there is nothing more that we can do here? Would really like to be able to get this working on all my machines.

**Comment by JZXPacky** [comment]
So one thing I've noticed is that my PiZeroW's all had the same machine-id and also the same "Pretty Hostname" in /etc/machine-info as they were all cloned from the same card. I've gone and fixed all these and now rebooted everything but still get the same error. I'm going to try and remove and re-install cockpit to see if that changes anything.

**Comment by JZXPacky** [comment]
Ok that didn't work. Same result.

**Comment by JZXPacky** [comment]
Is there a way I can get around the remotectl issue? I am using my own self signed ssl cert for my installs.

**Comment by JZXPacky** [comment]
Ok I tried changing that and it made no difference, so I ran a stack trace on the ExecStart command and got this.
```
gdb --args /usr/lib/cockpit/cockpit-ws
GNU gdb (Raspbian 7.7.1+dfsg-5+rpi1) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/lib/cockpit/cockpit-ws...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/lib/cockpit/cockpit-ws
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x0006778e in ?? ()
(gdb) bt full
#0  0x0006778e in ?? ()
No symbol table info available.
#1  0x000185da in ?? ()
No symbol table info available.
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb)
```

**Comment by wembleyford** [comment]
Just wanted to chip in on this and observe that I am having exactly the same issue on a Rasperry Pi B rev 1. The common factor between this and a Raspberry Pi zero is that they're both using the Arm 6 while the Pi 3 (and RPi B2) use an Arm 7 CPU.  

I'm running an Rpi B2 - and it's working fine with cockpit, but my Rpi B1 does not.

**Comment by JZXPacky** [comment]
@wembleyford Thanks for commenting with that, at least we have some idea now!

@martinpitt any chance of getting this to work on the ARMv6 devices?

**Comment by jwulf** [comment]
I built a armhf build on my RPi0w following these instructions: https://github.com/cockpit-project/cockpit/issues/7366#issuecomment-317665596 

```
# install build dependencies
sudo apt-get install -y debhelper dh-systemd dh-autoreconf autoconf automake git intltool libssh-dev zlib1g-dev libkrb5-dev libxslt1-dev libglib2.0-dev libsystemd-dev libpolkit-agent-1-dev libjson-glib-dev libpam0g-dev xsltproc xmlto docbook-xsl glib-networking openssh-client python

dget http://httpredir.debian.org/debian/pool/main/c/cockpit/cockpit_176-1.dsc
dpkg-source -x cockpit_*.dsc
cd cockpit-*
dpkg-buildpackage -us -uc -b
```

**Comment by fthiery** [comment]
For what it's worth, i'm replicating this behaviour on amd64 architecture on a yocto build.

```(gdb) run certificate --ensure --user=root --group= --selinux-type=
Starting program: /usr/sbin/remotectl certificate --ensure --user=root --group= --selinux-type=
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/libthread_db.so.1".
[Detaching after fork from child process 1018]
[Detaching after fork from child process 1019]

** (remotectl:1014): WARNING **: 09:54:29.494: (null)

Program received signal SIGSEGV, Segmentation fault.
0x000055555555965c in ?? ()
(gdb) bt full
#0  0x000055555555965c in ?? ()
No symbol table info available.
#1  0x0000555555557a50 in ?? ()
No symbol table info available.
#2  0x00007ffff7a5108b in __libc_start_main (main=0x5555555577b0, argc=6, argv=0x7fffffffec18, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffec08) at ../csu/libc-start.c:308
        self = <optimized out>
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {0, 5860336419568800779, 93824992246384, 140737488350224, 0, 0, 288597303926641675, 288614165598753803}, mask_was_saved = 0}}, priv = {
            pad = {0x0, 0x0, 0x7fffffffec50, 0x7ffff7ffe130}, data = {prev = 0x0, cleanup = 0x0, canceltype = -5040}}}
        not_first_call = <optimized out>
#3  0x0000555555557a9a in ?? ()
No symbol table info available.```

**Comment by fthiery** [comment]
Here is the output of strace -f

It seems that the subprocess call to openssl fails

```
[pid 32400] execve("/sbin/openssl", ["openssl", "req", "-x509", "-days", "36500", "-newkey", "rsa:2048", "-keyout", "/etc/cockpit/ws-certs.d/0-self-s"..., "-keyform", "PEM", "-nodes", "-out", "/etc/cockpit/ws-certs.d/0-self-s"..., "-outform", "PEM", "-subj", "/O=dd638955b9404cb6aa8190ffe9223"..., "-config", "/tmp/ssl.conf.TT2HPZ", "-extensions", "v3_req"], 0x55c0e3452270 /* 19 vars */) = -1 ENOENT (No such file or directory)
```

[strace_segfault.log](https://github.com/cockpit-project/cockpit/files/2383278/strace_segfault.log)

**Comment by fthiery** [comment]
Oh, no /sbin/openssl on my system. Maybe it should not segfault like that in that case...

 @JZXPacky what does the strace give you ?

**Comment by fthiery** [comment]
Being on yocto, i had to install the openssl-bin package to include the /sbin/openssl binary; now remotectl does not segfault at all.

This seems to mean that this ticket should be renamed: remotectl segfaults when it does not find the openssl binary, which indicates that the error handling is not clean at https://github.com/cockpit-project/cockpit/blob/cb29836c3c048f024e7062b10585a9b7f1abf99f/src/ws/cockpitcertificate.c#L174

**Comment by martinpitt** [comment]
Nice, thanks @fthiery! I tried this on current version 177, and remotectl does crash without /bin/openssl.

**Comment by martinpitt** [comment]
Fixed in PR #10107. Thanks for your report!

**Comment by JZXPacky** [comment]
Hi all, sorry I haven't commented recently, just started a new job and have been rather busy tidying up the network there.

Awesome news that there is a fix, are we still installing using the same original install instructions or are we compiling from source? I just tried installing using the backports on both Jessie and Stretch and get the same result. Excuse my questions, it's been a while since I've looked at this and forgotten how I did it in the first place.

**Comment by martinpitt** [comment]
@JZXPacky : If you just install the "openssl" package, things should work with the current cockpit version. This fix will just make sure that (1) remotectl doesn't crash, but  tells you about the missing openssl, and (2) add openssl as a dependency, so that it should always be available for cockpit.

So in summary, you can use the original install instructions.

**Comment by JZXPacky** [comment]
I tried installing openssl and cockpit again but getting the same result. I've checked and my openssl is located in /usr/bin/openssl.

Where is cockpit looking for it?

Also, here is my strace.

```execve("/usr/sbin/remotectl", ["/usr/sbin/remotectl", "certificate", "--ensure", "--user=root", "--group=", "--selinux-type="], [/* 38 vars */]) = 0
brk(0)                                  = 0x1708000
uname({sys="Linux", node="PackyPiZeroW01", ...}) = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap2(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f9b000
access("/etc/ld.so.preload", R_OK)      = 0
open("/etc/ld.so.preload", O_RDONLY|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=42, ...}) = 0
mmap2(NULL, 42, PROT_READ|PROT_WRITE, MAP_PRIVATE, 3, 0) = 0xb6f9a000
close(3)                                = 0
open("/usr/lib/arm-linux-gnueabihf/libarmmem.so", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0h\5\0\0004\0\0\0"..., 512) = 512
lseek(3, 17960, SEEK_SET)               = 17960
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 960) = 960
lseek(3, 17696, SEEK_SET)               = 17696
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\3\f\1\22\4\24"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=18920, ...}) = 0
mmap2(NULL, 83236, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6f59000
mprotect(0xb6f5e000, 61440, PROT_NONE)  = 0
mmap2(0xb6f6d000, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x4000) = 0xb6f6d000
mprotect(0xbec97000, 4096, PROT_READ|PROT_WRITE|PROT_EXEC|PROT_GROWSDOWN) = 0
close(3)                                = 0
munmap(0xb6f9a000, 42)                  = 0
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=84390, ...}) = 0
mmap2(NULL, 84390, PROT_READ, MAP_PRIVATE, 3, 0) = 0xb6f44000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/arm-linux-gnueabihf/libgio-2.0.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\260k\2\0004\0\0\0"..., 512) = 512
lseek(3, 1290932, SEEK_SET)             = 1290932
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1080) = 1080
lseek(3, 1290584, SEEK_SET)             = 1290584
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=1292012, ...}) = 0
mmap2(NULL, 1360044, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6df7000
mprotect(0xb6f2f000, 65536, PROT_NONE)  = 0
mmap2(0xb6f3f000, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x138000) = 0xb6f3f000
mmap2(0xb6f43000, 172, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb6f43000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/arm-linux-gnueabihf/libgobject-2.0.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0 \207\0\0004\0\0\0"..., 512) = 512
lseek(3, 300640, SEEK_SET)              = 300640
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1040) = 1040
lseek(3, 300304, SEEK_SET)              = 300304
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=301680, ...}) = 0
mmap2(NULL, 367268, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6d9d000
mprotect(0xb6de6000, 61440, PROT_NONE)  = 0
mmap2(0xb6df5000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x48000) = 0xb6df5000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libglib-2.0.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0(T\1\0004\0\0\0"..., 512) = 512
lseek(3, 993420, SEEK_SET)              = 993420
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1080) = 1080
lseek(3, 993072, SEEK_SET)              = 993072
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=994500, ...}) = 0
mmap2(NULL, 1060192, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6c9a000
mprotect(0xb6d8c000, 61440, PROT_NONE)  = 0
mmap2(0xb6d9b000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xf1000) = 0xb6d9b000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libsystemd.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\2502\0\0004\0\0\0"..., 512) = 512
lseek(3, 131460, SEEK_SET)              = 131460
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1160) = 1160
lseek(3, 131088, SEEK_SET)              = 131088
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=132620, ...}) = 0
mmap2(NULL, 198720, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6c69000
mprotect(0xb6c88000, 65536, PROT_NONE)  = 0
mmap2(0xb6c98000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1f000) = 0xb6c98000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0lY\0\0004\0\0\0"..., 512) = 512
lseek(3, 120788, SEEK_SET)              = 120788
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1520) = 1520
lseek(3, 86420, SEEK_SET)               = 86420
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0755, st_size=122308, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f9a000
mmap2(NULL, 160316, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6c41000
mprotect(0xb6c55000, 65536, PROT_NONE)  = 0
mmap2(0xb6c65000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x14000) = 0xb6c65000
mmap2(0xb6c67000, 4668, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb6c67000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0L\204\1\0004\0\0\0"..., 512) = 512
lseek(3, 1239936, SEEK_SET)             = 1239936
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 2840) = 2840
lseek(3, 1236500, SEEK_SET)             = 1236500
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0755, st_size=1242776, ...}) = 0
mmap2(NULL, 1312152, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6b00000
mprotect(0xb6c2b000, 65536, PROT_NONE)  = 0
mmap2(0xb6c3b000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x12b000) = 0xb6c3b000
mmap2(0xb6c3e000, 9624, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb6c3e000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/arm-linux-gnueabihf/libgmodule-2.0.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0 \r\0\0004\0\0\0"..., 512) = 512
lseek(3, 12824, SEEK_SET)               = 12824
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1000) = 1000
lseek(3, 12500, SEEK_SET)               = 12500
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=13824, ...}) = 0
mmap2(NULL, 78068, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6aec000
mprotect(0xb6aef000, 61440, PROT_NONE)  = 0
mmap2(0xb6afe000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0xb6afe000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0l\30\0\0004\0\0\0"..., 512) = 512
lseek(3, 90768, SEEK_SET)               = 90768
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1120) = 1120
lseek(3, 90404, SEEK_SET)               = 90404
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=91888, ...}) = 0
mmap2(NULL, 155944, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6ac5000
mprotect(0xb6adb000, 61440, PROT_NONE)  = 0
mmap2(0xb6aea000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x15000) = 0xb6aea000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libselinux.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\364G\0\0004\0\0\0"..., 512) = 512
lseek(3, 111680, SEEK_SET)              = 111680
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1080) = 1080
lseek(3, 111396, SEEK_SET)              = 111396
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=112760, ...}) = 0
mmap2(NULL, 183460, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6a98000
mprotect(0xb6ab3000, 61440, PROT_NONE)  = 0
mmap2(0xb6ac2000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a000) = 0xb6ac2000
mmap2(0xb6ac4000, 3236, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb6ac4000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libresolv.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0@%\0\0004\0\0\0"..., 512) = 512
lseek(3, 74444, SEEK_SET)               = 74444
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1200) = 1200
lseek(3, 74096, SEEK_SET)               = 74096
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=75644, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f99000
mmap2(NULL, 149428, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6a73000
mprotect(0xb6a85000, 61440, PROT_NONE)  = 0
mmap2(0xb6a94000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x11000) = 0xb6a94000
mmap2(0xb6a96000, 6068, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb6a96000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/arm-linux-gnueabihf/libffi.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\204\23\0\0004\0\0\0"..., 512) = 512
lseek(3, 29256, SEEK_SET)               = 29256
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1120) = 1120
lseek(3, 28932, SEEK_SET)               = 28932
read(3, "A,\0\0\0aeabi\0\1\"\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 45) = 45
fstat64(3, {st_mode=S_IFREG|0644, st_size=30376, ...}) = 0
mmap2(NULL, 95040, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6a5b000
mprotect(0xb6a62000, 61440, PROT_NONE)  = 0
mmap2(0xb6a71000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0xb6a71000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libpcre.so.3", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0004\20\0\0004\0\0\0"..., 512) = 512
lseek(3, 401744, SEEK_SET)              = 401744
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1040) = 1040
lseek(3, 401436, SEEK_SET)              = 401436
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=402784, ...}) = 0
mmap2(NULL, 467028, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb69e8000
mprotect(0xb6a4a000, 61440, PROT_NONE)  = 0
mmap2(0xb6a59000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x61000) = 0xb6a59000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/librt.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\3\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0000\30\0\0004\0\0\0"..., 512) = 512
lseek(3, 25352, SEEK_SET)               = 25352
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1320) = 1320
lseek(3, 24964, SEEK_SET)               = 24964
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=26672, ...}) = 0
mmap2(NULL, 90684, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb69d1000
mprotect(0xb69d7000, 61440, PROT_NONE)  = 0
mmap2(0xb69e6000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x5000) = 0xb69e6000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/liblzma.so.5", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0H\37\0\0004\0\0\0"..., 512) = 512
lseek(3, 123512, SEEK_SET)              = 123512
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1080) = 1080
lseek(3, 123224, SEEK_SET)              = 123224
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=124592, ...}) = 0
mmap2(NULL, 155996, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb69aa000
mprotect(0xb69c8000, 28672, PROT_NONE)  = 0
mmap2(0xb69cf000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1d000) = 0xb69cf000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libgcrypt.so.20", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\0O\0\0004\0\0\0"..., 512) = 512
lseek(3, 598204, SEEK_SET)              = 598204
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1120) = 1120
lseek(3, 597844, SEEK_SET)              = 597844
read(3, "A,\0\0\0aeabi\0\1\"\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 45) = 45
fstat64(3, {st_mode=S_IFREG|0644, st_size=599324, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f98000
mmap2(NULL, 664140, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb6907000
mprotect(0xb6995000, 61440, PROT_NONE)  = 0
mmap2(0xb69a4000, 20480, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x8d000) = 0xb69a4000
mmap2(0xb69a9000, 588, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb69a9000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\220\t\0\0004\0\0\0"..., 512) = 512
lseek(3, 8660, SEEK_SET)                = 8660
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1160) = 1160
lseek(3, 8328, SEEK_SET)                = 8328
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=9820, ...}) = 0
mmap2(NULL, 73912, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb68f4000
mprotect(0xb68f6000, 61440, PROT_NONE)  = 0
mmap2(0xb6905000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1000) = 0xb6905000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libgcc_s.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0`\321\0\0004\0\0\0"..., 512) = 512
lseek(3, 116400, SEEK_SET)              = 116400
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1120) = 1120
lseek(3, 116072, SEEK_SET)              = 116072
read(3, "A.\0\0\0aeabi\0\1$\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\24\1\25"..., 47) = 47
fstat64(3, {st_mode=S_IFREG|0644, st_size=117520, ...}) = 0
mmap2(NULL, 181664, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb68c7000
mprotect(0xb68e4000, 61440, PROT_NONE)  = 0
mmap2(0xb68f3000, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1c000) = 0xb68f3000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/arm-linux-gnueabihf/libgpg-error.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0(\0\1\0\0\0\340\34\0\0004\0\0\0"..., 512) = 512
lseek(3, 53840, SEEK_SET)               = 53840
read(3, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 1080) = 1080
lseek(3, 53544, SEEK_SET)               = 53544
read(3, "A0\0\0\0aeabi\0\1&\0\0\0\0056\0\6\6\10\1\t\1\n\2\22\4\23\1\24"..., 49) = 49
fstat64(3, {st_mode=S_IFREG|0644, st_size=54920, ...}) = 0
mmap2(NULL, 119120, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xb68a9000
mprotect(0xb68b6000, 61440, PROT_NONE)  = 0
mmap2(0xb68c5000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0xb68c5000
close(3)                                = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f97000
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f96000
set_tls(0xb6f964c0, 0xb6f96c58, 0xb6f9e058, 0xb6f964c0, 0xb6f9e058) = 0
mprotect(0xb6c3b000, 8192, PROT_READ)   = 0
mprotect(0xb68c5000, 4096, PROT_READ)   = 0
mprotect(0xb6905000, 4096, PROT_READ)   = 0
mprotect(0xb69a4000, 4096, PROT_READ)   = 0
mprotect(0xb69cf000, 4096, PROT_READ)   = 0
mprotect(0xb6c65000, 4096, PROT_READ)   = 0
mprotect(0xb69e6000, 4096, PROT_READ)   = 0
mprotect(0xb6a59000, 4096, PROT_READ)   = 0
mprotect(0xb6a71000, 4096, PROT_READ)   = 0
mprotect(0xb6a94000, 4096, PROT_READ)   = 0
mprotect(0xb6ac2000, 4096, PROT_READ)   = 0
mprotect(0xb6aea000, 4096, PROT_READ)   = 0
mprotect(0xb6d9b000, 4096, PROT_READ)   = 0
mprotect(0xb6afe000, 4096, PROT_READ)   = 0
mprotect(0xb6c98000, 4096, PROT_READ)   = 0
mprotect(0xb6df5000, 4096, PROT_READ)   = 0
mprotect(0xb6f3f000, 8192, PROT_READ)   = 0
mprotect(0xb6f59000, 20480, PROT_READ|PROT_WRITE) = 0
mprotect(0xb6f59000, 20480, PROT_READ|PROT_EXEC) = 0
cacheflush(0xb6f59000, 0xb6f5e000, 0, 0x15, 0xbec95f08) = 0
mprotect(0x24000, 4096, PROT_READ)      = 0
mprotect(0xb6f9d000, 4096, PROT_READ)   = 0
munmap(0xb6f44000, 84390)               = 0
set_tid_address(0xb6f96068)             = 6002
set_robust_list(0xb6f96070, 12)         = 0
rt_sigaction(SIGRTMIN, {0xb6c46434, [], SA_RESTORER|SA_SIGINFO, 0xb6b2f1a0}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0xb6c462d8, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0xb6b2f1a0}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=8192*1024, rlim_max=RLIM_INFINITY}) = 0
brk(0)                                  = 0x1708000
brk(0x1729000)                          = 0x1729000
statfs64("/sys/fs/selinux", 88, 0xbec97270) = -1 ENOENT (No such file or directory)
statfs64("/selinux", 88, 0xbec97270)    = -1 ENOENT (No such file or directory)
open("/proc/filesystems", O_RDONLY|O_LARGEFILE) = 3
fstat64(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xb6f95000
read(3, "nodev\tsysfs\nnodev\trootfs\nnodev\tr"..., 1024) = 361
read(3, "", 1024)                       = 0
close(3)                                = 0
munmap(0xb6f95000, 4096)                = 0
gettimeofday({1537432578, 894014}, NULL) = 0
futex(0xb6d9cca0, FUTEX_WAKE, 2147483647) = 0
futex(0xb6d9cca0, FUTEX_WAKE, 2147483647) = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x61782} ---
+++ killed by SIGSEGV +++
Segmentation fault

**Comment by fthiery** [comment]
Oops look like i jumped the gun: this seems to be a different issue after all, sorry. @martinpitt should JZXPacky create a new ticket ?

@JZXPacky can you do the following:

```
gdb /usr/sbin/remotectl
(gdb) run certificate --ensure --user=root --group= --selinux-type=
when it crashes
(gdb) bt full
```

**Comment by JZXPacky** [comment]
@fthiery Ok here we go.

```
gdb /usr/sbin/remotectl
GNU gdb (Raspbian 7.7.1+dfsg-5+rpi1) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/sbin/remotectl...(no debugging symbols found)...done.
(gdb) run certificate --ensure --user=root --group= --selinux-type=
Starting program: /usr/sbin/remotectl certificate --ensure --user=root --group= --selinux-type=
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00061782 in ?? ()
(gdb) bt full
#0  0x00061782 in ?? ()
No symbol table info available.
#1  0x000125ce in ?? ()
No symbol table info available.
Backtrace stopped: previous frame identical to this frame (corrupt stack?)

**Comment by fthiery** [comment]
Well, not that useful. You probably need to recompile it in debug mode to get a useful trace. This will probably take a while on your pi :)

```
apt update
apt install git nodejs npm gdb
apt build-dep cockpit
git clone https://github.com/cockpit-project/cockpit.git
cd cockpit
./autogen.sh --prefix=/usr --enable-debug --disable-pcp
make
gdb ./remotectl
(gdb) run certificate --ensure --user=root --group= --selinux-type=
```

**Comment by clovenguth** [comment]
Hello I just wanted to add here if it helps any of you what I've experienced with 3 Pi Zero Ws running the latest Debian 9.4 in case it helps. I had just come across Cockpit 2 days ago and successfully installed it on a couple Pi 3s running 9.4. All my Pi Zeros while installing through ssh would drop connection at the point of "setting up modemmanager (1.6.4-1)"  because the Pi zeros are changing their MAC addresses and I have my Pi's IPs reserved with MAC bindings. It's not until I remove Cockpit, and then do an apt-get autoremove does my Pi return to the correct mac address.

Sorry I don't have much time to do any debugging myself, and I don't know if above helps any of you at all, but just wanted to share.

**Comment by fthiery** [comment]
@clovenguth seems unrelated and needs more digging i think
@JZXPacky really sorry, i don't know how the package is being built. You may need to do apt source cockpit inside the PI, and add the --enable-debug flag somewhere, then rebuild and install the package. Then you should be able to trace the error.

**Comment by JZXPacky** [comment]
@fthiery I followed your instructions you posted above and that's how I got that error.

```
apt update
apt install git nodejs npm gdb
apt build-dep cockpit
git clone https://github.com/cockpit-project/cockpit.git
cd cockpit
./autogen.sh --prefix=/usr --enable-debug --disable-pcp
make

When I executed make is when I got the error I posted. I'm going to try again as the SD card died on that particular Pi Zero.

I'm looking at the errors with react-lite and trying to solve those to see if that fixes the issues with make.

**Comment by JZXPacky** [comment]
No love with this, seems to be related to React.

`Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration.
